### PR TITLE
Add feature gate to enable master to cluster traffic

### PR DESCRIPTION
**Pull request by irozzo-1A**
This PR introduces the feature gate to enable the master to cluster traffic.

Despite this does not represent a very significant increase in the implementation of the functionality I created a dedicated PR as this will be used both on the agent and the server side.

Ref: #171

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (cheftako) [inline pkg/features/features.go:2]: 2021

**Comment by cheftako** [comment]
cc @deads2k 
Reminded of our conversation https://github.com/kubernetes/kubernetes/pull/92201#discussion_r452370878. I think even here there is something to be said for this feature being Kubernetes version bound rather than ANP version bound. It would suggest we might want this feature in k8s.io/api.

**Comment by deads2k** [comment]
> cc @deads2k
> Reminded of our conversation [kubernetes/kubernetes#92201 (comment)](https://github.com/kubernetes/kubernetes/pull/92201#discussion_r452370878). I think even here there is something to be said for this feature being Kubernetes version bound rather than ANP version bound. It would suggest we might want this feature in k8s.io/api.

I think having featuregate names in k8s.io/API is a good idea, but I think I would draw a boundary for inclusion there at "used by a binary in k/k".  I think that having distinct featuregates for different binaries is pretty reasonable.

**Comment by cheftako** [comment]
cc @andrewsykim

**Pull request merged**